src/x3d/x3dload.pas

```diff
@@ -351,13 +334,78 @@ function LoadNode(const URL: string;
     Result := LoadGLTF(URL)
   else
 
+  if (MimeType = 'image/starling-texture-atlas') or
+     (URIMimeType(URIDeleteAnchor(URL, true)) = 'image/starling-texture-atlas') then
+    Result := LoadStarlingTextureAtlas(URL)
+  else
+
+  if (MimeType = 'application/x-plist') or
+     (URIMimeType(URIDeleteAnchor(URL, true)) = 'application/x-plist') then
+    Result := LoadCocos2d(URL)
+  else
+
+  { Support for simple graphics images like PNG }
+  if URIMimeExtensions.ContainsKey(ExtractFileExt(URIDeleteAnchor(URL, true))) then
+    Result := LoadImageAsX3DModel(URL)
+  else
+
   if NilOnUnrecognizedFormat then
     Result := nil
   else
     raise Exception.CreateFmt('Unrecognized file type "%s" for scene "%s"',
       [MimeType, URIDisplay(URL)]);
 end;
 
+function LoadScene_FileFilters: String;
+
+  function ReadImageExtensions: String;
```

Comment:
Instead of this, use `AllImagesExtensions` proposed above.

P.S. For constructions like below, the `SAppendPart` is useful.